packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -586,29 +588,55 @@ async function handler(
   const contactOwnerEmail = skipContactOwner ? null : contactOwnerFromReq;
 
   let routingFormResponse = null;
-
   if (routedTeamMemberIds) {
     //routingFormResponseId could be 0 for dry run. So, we just avoid undefined value
     if (routingFormResponseId === undefined) {
       throw new HttpError({ statusCode: 400, message: "Missing routingFormResponseId" });
     }
-    routingFormResponse = await prisma.app_RoutingForms_FormResponse.findUnique({
-      where: {
-        id: routingFormResponseId,
-      },
-      select: {
-        response: true,
-        form: {
-          select: {
-            routes: true,
-            fields: true,
+
+    if (queuedFormResponse) {
+      // Write to the routing form response table and return the result
+      routingFormResponse = await RoutingFormResponseRepository.writeQueuedFormResponseToFormResponse(
+        routingFormResponseId,
+        bookerEmail
+      );
+      if (!routingFormResponse) {
+        throw new HttpError({ statusCode: 400, message: "Missing routingFormResponse" });
+      }
+      routingFormResponseId = routingFormResponse.id;
+    } else {
+      routingFormResponse = await prisma.app_RoutingForms_FormResponse.findUnique({
+        where: {
+          id: routingFormResponseId,
+        },
+        select: {
+          response: true,
+          form: {
+            select: {
+              routes: true,
+              fields: true,
+            },
           },
+          chosenRouteId: true,
         },
-        chosenRouteId: true,
-      },
-    });
-  }
+      });
+    }
+  } else if (queuedFormResponse) {
```

Comment:
Even if there's no `routedTeamMemberIds` but a queued routing form response, we need to record the response.